pkg/iac/terraform/state/backend/tfcloud_reader.go

```diff
@@ -26,38 +28,79 @@ type TFCloudBody struct {
 	Data TFCloudData `json:"data"`
 }
 
-func NewTFCloudReader(client pkghttp.HTTPClient, workspaceId string, opts *Options) (*HTTPBackend, error) {
-	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/workspaces/%s/current-state-version", TFCloudAPI, workspaceId), nil)
-
-	if err != nil {
-		return nil, err
-	}
-
-	req.Header.Add("Content-Type", "application/vnd.api+json")
-	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", opts.TFCloudToken))
-
-	res, err := client.Do(req)
+type TFCloudBackend struct {
```

Comment:
Is it because of the way we use the term `reader` on the other files (file_reader, http_reader, s3_reader)? 

I could change `tfcloud_reader` to `tfcloud` and `tfcloud_config_reader` to `tfcloud_config`, wdyt ?

Reply:
Ok nervermind if other backend use this naming so. I didn't check how they were named ^^